android-client-sdk/src/main/java/com/devcycle/sdk/android/api/DVCClient.kt

```diff
@@ -98,6 +86,46 @@ class DVCClient private constructor(
         }
     }
 
+    private val onPauseApplication = fun () {
+        Timber.d("Closing streaming event source connection")
+        eventSource?.close()
+    }
+
+    private val onResumeApplication = fun () {
+        if (eventSource?.state != ReadyState.OPEN) {
+            eventSource?.close()
+            Timber.d("Restarting streaming event source connection")
+            initEventSource()
+            refetchConfig(false, null)
+        }
+    }
+
+    private fun initEventSource () {
+        if (config?.sse?.url != null) {
+            eventSource = EventSource.Builder(Handler(fun(messageEvent: MessageEvent?) {
+                if (messageEvent != null) {
```

Comment:
we can probably collapse these if statements, e.g. `if (messageEvent == null) { return }`